tools/metatoenv.py

```diff
@@ -0,0 +1,218 @@
+parser.add_argument('--channels', default='conda-forge')
+parser.add_argument('--platform', '--os', default='linux64')
+parser.add_argument('--extra', default='')
+parser.add_argument('--merge-with', default='')
+
+
+def _indentation_level(string):
+    """
+    Count the number of leading spaces in a string.
+    """
+    return len(string) - len(string.lstrip(' '))
+
+
+def _parse_yaml(text):
+    """
+    Parse yaml text and store into a dict.
+    Description of algorithm:
+     - if a line ends with `:`, we make a new nested dict. Current handle is then this
+       new nested dict.
+     - if a line does not end with `:`, but contains `:`, we set the value at the
+       current handle in the dict
+     - if a line does not end with `:`, and starts with `-`, we add the entry as a key
+       in the dict and set the value to `None`
+     - if the indentation level of the current line is less than the previous line, it
+       implies we a closing a block. We iterate backwards in the file until we find a
+       line with the same indent level and use the parent block of that line as our new
+       handle
+    """
+    out = {}
+    nspaces = 0
+    handle = out
+    path = []
+
+    # Remove all empty lines in text
+    clean_text = []
+    for line in text:
+        line = line.rstrip(' \n')
+        if len(line) > 0:
+            clean_text.append(line)
+
+    for i, line in enumerate(clean_text):
+        line = line.rstrip(' \n')
+
+        if i > 0:
+            line_indent = _indentation_level(line)
+            if line_indent < _indentation_level(clean_text[i - 1]):
+                current_indent = line_indent
+                for p in path[::-1]:
+                    if current_indent == p[1]:
+                        ind = path.index(p)
+                        handle = out
+                        for j in range(ind):
+                            handle = handle[path[j][0]]
+                        path = path[:ind]
+                        break
+
+        if line.endswith(':'):
+            key = line.strip(' -\n')
+            handle[key] = {}
+            handle = handle[key]
+            nspaces = _indentation_level(line)
+            path.append((key, nspaces))
+        else:
+            stripped = line.lstrip(' ')
+            if stripped.startswith('-'):
+                handle[stripped.strip(' -')] = None
+            elif ':' in stripped:
+                ind = stripped.find(':')
+                handle[stripped[:ind]] = stripped[ind + 1:]
+
+    return out
+
+
+def _jinja_filter(dependencies, platform):
+    """
+    Filter out deps for requested platform via jinja syntax.
+    """
+    out = {}
+    for key, value in dependencies.items():
+        if isinstance(value, dict):
+            out[key] = _jinja_filter(value, platform)
+        else:
+            ok = True
+            # Strip comments from key
+            key = key.split('#')[0].strip()
+            if key.count('[') > 0:
```

Comment:
Looks like this code (below) ignores any but the first pair of brackets. But I think this is fine, this tool is for developers and does not have to be super robust.